pkg/mpc-examples/contracts/03-ebu-rebalancer/EBURebalancerFactory.sol

```diff
@@ -0,0 +1,140 @@
+ * @notice Factory for a Managed Pool and EBURebalancer Controller.
+ * @dev Determines controller deployment address, deploys pool (w/ controller address as argument), then controller.
+ */
+contract EBURebalancerFactory is Ownable {
+    mapping(address => bool) public isControllerFromFactory;
+
+    address public immutable managedPoolFactory;
+    IVault public immutable balancerVault;
+    bool private _disabled;
+
+    uint256 private _nextControllerSalt;
+    address private _lastCreatedPool;
+
+    // This struct is a subset of IManagedPoolFactory.NewPoolParams which omits arguments
+    // that this factory will override and are therefore unnecessary to provide. It will
+    // ultimately be used to populate IManagedPoolFactory.NewPoolParams.
+    struct MinimalPoolParams {
+        string name;
+        string symbol;
+        IERC20[] tokens;
+        uint256[] normalizedWeights;
+        uint256 swapFeePercentage;
+        bool swapEnabledOnStart;
+        uint256 managementAumFeePercentage;
+        uint256 aumFeeId;
+    }
+
+    event ControllerCreated(address indexed controller, IVault vault, bytes32 poolId);
+    event Disabled();
+
+    constructor(IVault vault, address factory) {
+        balancerVault = vault;
+        managedPoolFactory = factory;
+    }
+
+    /**
+     * @dev Return the address of the most recently created pool.
+     */
+    function getLastCreatedPool() external view returns (address) {
+        return _lastCreatedPool;
+    }
+
+    function create(MinimalPoolParams memory minimalParams) external {
+        _ensureEnabled();
+
+        bytes32 controllerSalt = bytes32(_nextControllerSalt);
+        _nextControllerSalt += 1;
+
+        bytes memory controllerCreationCode = abi.encodePacked(
+            type(EBURebalancer).creationCode,
+            abi.encode(balancerVault)
+        );
+        address expectedControllerAddress = Create2.computeAddress(controllerSalt, keccak256(controllerCreationCode));
+
+        // Build arguments to deploy pool from factory.
+        address[] memory assetManagers = new address[](minimalParams.tokens.length);
+        for (uint256 i = 0; i < assetManagers.length; i++) {
+            assetManagers[i] = expectedControllerAddress;
+        }
+
+        // Populate IManagedPoolFactory.NewPoolParams with arguments from MinimalPoolParams and
+        // other arguments that this factory provides itself.
+        IManagedPoolFactory.NewPoolParams memory fullParams;
+        fullParams.name = minimalParams.name;
+        fullParams.symbol = minimalParams.symbol;
+        fullParams.tokens = minimalParams.tokens;
+        fullParams.normalizedWeights = minimalParams.normalizedWeights;
+        // Asset Managers set to the controller address, not known by deployer until creation.
+        fullParams.assetManagers = assetManagers;
+        fullParams.swapFeePercentage = minimalParams.swapFeePercentage;
+        fullParams.swapEnabledOnStart = minimalParams.swapEnabledOnStart;
```

Comment:
for this pool, it might make sense for the controller to set `swapEnabledOnStart = false`

Reply:
I agree